helm/prefect-server/templates/agent/deployment.yaml

```diff
@@ -0,0 +1,91 @@
+{{- if .Values.agent.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "prefect-server.nameField" (merge (dict "component" "agent") .) }}
+  labels: 
+    {{- include "prefect-server.commonLabels" . | nindent 4 }}
+    app.kubernetes.io/component: agent
+  annotations:
+    {{- merge .Values.agent.annotations .Values.annotations | toYaml | nindent 4 }}
+spec:
+  replicas: {{ .Values.agent.replicas }}
+  selector:
+    matchLabels:
+      {{- include "prefect-server.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: agent
+  template:
+    metadata:
+      labels:
+        {{- include "prefect-server.matchLabels" . | nindent 8 }}
+        app.kubernetes.io/component: agent
+    spec:
+      imagePullSecrets: {{ concat .Values.agent.image.pullSecrets .Values.imagePullSecrets | uniq | toJson }}
+      serviceAccountName: {{ include "prefect-server.serviceAccountName" . }}
+      {{- with .Values.agent.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+      - name: agent
+        {{- with .Values.agent.podSecurityContext }}
+        securityContext:
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+        image: "{{ .Values.agent.image.name }}:{{ .Values.agent.image.tag |  default .Values.prefectVersionTag }}"
+        imagePullPolicy: {{ .Values.agent.image.pullPolicy  }}
+        command:
+          - bash
+          - "-c"
+          - "prefect agent kubernetes start"
+        env:
+          - name: PREFECT__CLOUD__AGENT__AUTH_TOKEN
+            value: ''
```

Comment:
Is this env var necessary?  It's not strictly speaking wrong, but it might be confusing given that it doesn't have any effect w/ server

Reply:
I'm happy to remove it, I did the minimal changes necessary from the CLI generated yaml.